### ISSUE TITLE
Spotify blur effect is not re-applied when the application is restarted

### ISSUE THREAD

**Issue by ihsanauliaa** [esjeon/kwin-forceblur#11]
Hello, before all, thank you for your work!

I have successfully applied the force blur for Spotify. Unfortunately, I noticed that the effect is not reapplied when Spotify is restarted. Is there any workaround for this issue?

Thanks!

**Comment by esjeon** [comment]
Hmm, how did you make Spotify transparent? AFAIK, it doesn't natively support background transparency. If you used KWin window rules to apply transparency, it's possible that the rules are not correctly set.

If the problem persists, you can check if the script is working or not by checking window properties. Run `xprop _KDE_NET_WM_BLUR_BEHIND_REGION` in Konsole, and click the window of your interest. It should show something.

**Comment by ihsanauliaa** [comment]
Yeah I used KWin window rule and unchecked and checked the option to enable Force blur script and it works when the application is running, but it is not reapplied when I restarted spotify.

When I run `xprop _KDE_NET_WM_BLUR_BEHIND_REGION` after I uncheked and checked the option to enable force blur, it detected the blur properties (`0, 0, 1602, 933`) but when I restarted spotify, it said that it is not found.

I observed that when I launched spotify for the first time, its window size kind of flickered (the window started small, and then it resized to its proper size, probably beacuse it is loading something?). Maybe force blur doesn't detect the window because of it? I don't really know though

**Comment by esjeon** [comment]
I've just reproduced the bug on my environment. A weird thing is that the Spotify window has no "class" on startup. At least, that's what debug prints are saying. It's possible that window class and hints are added manually after creating the window.

I tried some quick workarounds, which didn't work. It may take some time.

**Comment by ihsanauliaa** [comment]
So that's the problem! If that's the case then that's unfortunate.

Thank you anyway!

**Comment by RitwickVerma** [comment]
A workaround is to leave a blank line in the list of window class names.

**Comment by ihsanauliaa** [comment]
@RitwickVerma can you elaborate? How can I leave a blank line in the list of window class names? Through settings?

**Comment by RitwickVerma** [comment]
Yes. There should be a configure button next to the effect in kwin settings.
![image](https://user-images.githubusercontent.com/16993844/91941873-d0d05400-ecae-11ea-9fc9-bfe034095985.png)

Here's what my list looks like.
![image](https://user-images.githubusercontent.com/16993844/91941804-a8e0f080-ecae-11ea-85c2-0f24c533a61d.png)

**Comment by ihsanauliaa** [comment]
It does work! Thank you! I didn't realize that blank line is a valid window class name

**Comment by esjeon** [comment]
@RitwickVerma So, because the class name is initially empty, you just chose to match the empty class name. lol

**Comment by RitwickVerma** [comment]
haha yeah tbh I didnt expect it to work.

**Comment by diogofd8** [comment]
@RitwickVerma @esjeon 
What is the solution for when you mark "blur all except matching"? I wanted to avoid manually naming everything I want to blur but apparently even with an empty exceptions list it fails to blur spotify.
